main.ml

```diff
@@ -1197,7 +1197,13 @@ let self_test ~root_dir =
     Test.SR.stat ~dbg ~sr >>= fun _sr_info ->
     Test.SR.scan ~dbg ~sr >>= fun _sr_list ->
     return ()
-  ) >>= function
+  )
+
+let self_test ~root_dir =
+  (self_test_plugin ~root_dir "org.xen.xapi.storage.dummy"
```

Comment:
Why you need these parentheses?

Reply:
I want to match on the deferred that is the output of both `self_test_plugin` calls. Without the parenthesis the types don't match:
```
File "xapi-storage-script/main.ml", line 1322, characters 12-97:
Error: This expression has type
         unit Async_kernel__Deferred.t = unit Async_kernel__Types.Deferred.t
       but an expression was expected of type
         ('a, exn) Core_kernel.Result.t Async.Monitor.Deferred.t =
           ('a, exn) Core_kernel.Result.t Async_kernel__Types.Deferred.t
       Type unit is not compatible with type
         ('a, exn) Core_kernel.Result.t = ('a, exn) result 
make: *** [Makefile:7: release] Error 1
```

Note that `>>>=` is the bind for `Result.t Deferred.t` and `>>=` just the bind for `Deferred.t`

Reply:
It can be a bit confusing as this is different from `Lwt`. Although it is nice that I can match on the result without doing a try/catch.